tests/lib/rules/require-v-if-else-key.js

```diff
@@ -0,0 +1,124 @@
+/**
+ * @author Doug Wade <douglas.b.wade@gmail.com>
+ * See LICENSE file in root directory for full license.
+ */
+'use strict'
+
+const RuleTester = require('eslint').RuleTester
+const rule = require('../../../lib/rules/require-v-if-else-key')
+
+const tester = new RuleTester({
+  parser: require.resolve('vue-eslint-parser'),
+  parserOptions: {
+    ecmaVersion: 2020,
+    sourceType: 'module'
+  }
+})
+
+tester.run('require-v-if-else-key', rule, {
+  valid: [
+    {
+      filename: 'test.vue',
+      code: `
+      <template>
+        <div>
+          <div v-if="conditional" :key="foo">foo</div>
+          <div v-else :key="bar">bar</div>
+        </div>
+      </template>
+      `
+    },
+    {
+      filename: 'test.vue',
+      code: `
+      <template>
+        <div>
+          <div v-if="conditional" key="foo">foo</div>
+          <div v-else key="bar">bar</div>
+        </div>
+      </template>
+      `
+    },
+    {
+      filename: 'test.vue',
+      code: `
+      <template>
+        <div>
+          <div v-if="conditional">foo</div>
+        </div>
+      </template>
+      `
+    },
+    {
+      filename: 'test.vue',
+      code: `
+      <template>
+        <div>
+          <div v-if="conditional">foo</div>
+          <span v-else>bar</span>
+        </div>
+      </template>
+      `
+    },
+    {
+      filename: 'test.vue',
+      code: `
+      <template>
+        <div>
+          <div v-if="conditional">foo</div>
+          <span v-else-if="conditional">bar</span>
+        </div>
+      </template>
+      `
+    },
+    {
+      filename: 'test.vue',
+      code: `
+      <template>
+        <div>
+          <div v-if="conditional" key="foo">foo</div>
+          <div v-else-if="conditional" key="bar">bar</span>
+        </div>
+      </template>
+      `
+    }
```

Comment:
We need to add tests using `<template v-else>`. Note that vue2 does not allow you to place a key in `<template>`.